### PR TITLE
feat: network waterfall view for session replay

### PR DESCRIPTION
**Pull request by pauldambra**
#TravelHackathon

We have all the data we need to generate page reports in individual recording's session replay data

We can't make it queryable but we can display it individually, that's already useful so let's do that

Screenshot 2024-05-19 at 22 16 19

## TODO 

- [x] react to playlist ID changes
- [x] show breakdown of asset types by size
- [ ] allow filtering to remove asset types from view
- [ ] make it exportable?
- [ ] make it work with notebooks
- [ ] change the URL when e.g. paging so you can share with others
- [ ] allow public sharing?
- [ ] make an aggregate view that shows the whole session collapsed by URL with box-plots

## if it's a mode of the player we can ignore

- [ ] don't mark as viewed?

### CODE REVIEW

**Comment by pauldambra** [comment]
@daibhin thought I'd put something up you can play with to get UX feedback - am already thinking it should be called "page report" not "network view"

**Comment by pauldambra** [comment]
pulled out 

* https://github.com/PostHog/posthog/pull/22355
* https://github.com/PostHog/posthog/pull/22352
*

**Comment by daibhin** [comment]
This feels cool. Great to give people another means of analysing their data. I made two larger changes:
- Put the contents into a table for easier formatting
- Cleaned up the asset proportions to hide text on smaller screens

I think the main question I took away from it all was why this is a separate tab to the recordings? I'd imagine it would be quite annoying to have to switch between the two tabs and find a recording again if you want to look at the waterfall view.

Do we expect someone to use this while analysing a recording or is it something they'd intentionally come to do? Could this be a different view on the existing recording that you can toggle to? Or maybe it could just be another tab in the inspector?

**Comment by pauldambra** [comment]
> I think the main question I took away from it all was why this is a separate tab to the recordings?

Yeah, I think eventually you'd want to be able to visit individual page reports, share them, export them. 

It was separate mostly just cos I was figuring it out as I go. And a way to promote the feature more easily.

---

I don't think it should live _only_ in the inspector. It feels too restricted.

----

I'd be keen to get it in behind a flag so we can see the display with some production data to sense check that.

**Comment by pauldambra** [comment]
super confused why moving the initiator type color mapping has broken all uses of color in js tests 🫠

File: frontend/src/scenes/session-recordings/apm/NetworkView.tsx [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+import { PerformanceCardRow } from 'scenes/session-recordings/apm/components/PerformanceCard'
+import { MethodTag, StatusTag } from 'scenes/session-recordings/apm/playerInspector/ItemPerformanceEvent'
+import { NetworkBar } from 'scenes/session-recordings/apm/waterfall/NetworkBar'
+
+import { PerformanceEvent } from '~/types'
+
+import { networkViewLogic } from './networkViewLogic'
+
+function SimpleURL({ name, entryType }: { name: string | undefined; entryType: string | undefined }): JSX.Element {
+    // TODO we should show hostname if it isn't the same as the navigation for the page(s) we're looking at
```
Comment (marandaneto) [inline frontend/src/scenes/session-recordings/apm/NetworkView.tsx:21]: Not a must for this PR I believe, right?

Comment (pauldambra) [reply frontend/src/scenes/session-recordings/apm/NetworkView.tsx:21]: right, it's behind a flag... definitely should before wider release 👍

**Review by marandaneto — APPROVED** [review]
low context approval given :D since it's behind a flag anyways

**Pull request merged**